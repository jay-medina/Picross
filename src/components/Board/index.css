.row {
  display: flex;
  justify-content: flex-end;
}

.columns {
  display: flex;
  align-items: flex-end;
  justify-content: flex-end;
}
