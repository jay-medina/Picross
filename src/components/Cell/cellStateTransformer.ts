export enum CellState {
  Selected = 'selected',
  Crossed =  'crossed',
}

export interface CellStates {
  [key: string]: (CellState | undefined);
}


export function updateState(oldState: CellStates, row: number, column: number) {
  const key = createKey(row, column);
  const state = getState(oldState, key);

  return {
    ...oldState,
    [key]: toggleState(
      state,
      row,
      column,
    ),
  };
}

export function getState(state: CellStates, key: string) {
  return state[key];
}

export function createKey(row: number, column: number) {
  return `${row} - ${column}`;
}

function toggleState(state: CellState | undefined, row: number, column: number) {
  switch (state) {
    case CellState.Selected: return CellState.Crossed;
    case CellState.Crossed: return undefined;
    default: return CellState.Selected;
  }
}
