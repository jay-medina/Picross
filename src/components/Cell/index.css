.cell {
  width: 2.5rem;
  height: 2.5rem;
  border: 1px solid;
  border-color: var(--dark-blue);
  font-size: 1rem;
  display: flex;
  justify-content: center;
  background-color: var(--board-color);
  cursor: pointer;
}

.cell .selected {
  background-color: var(--white);
  margin: 0.15rem;
  border-radius: 3px;
  width: inherit;
}

.cell .cross {
  color: var(--secondary-color);
  font-size: 2rem;
}

.cell.hint {
  font-size: 1.35rem;
  padding-top: 0.25rem;
  color:white;
  background-color: var(--secondary-color);
}

.cell.border-cell-left {
  border-right-color: white;
}

.cell.border-cell-right {
  border-left-color: white;
}
