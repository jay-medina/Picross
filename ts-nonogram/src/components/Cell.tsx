import React from 'react';

const Cell: React.StatelessComponent<{}> = () => {
  return <div>Hello world</div>;
};

export default Cell;
