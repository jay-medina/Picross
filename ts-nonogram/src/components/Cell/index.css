:root {
  --main-color: #546e7a;
  --secondary-color: #29434e;
  --white: #FFF;
}
.cell {
  width: 2.5rem;
  height: 2.5rem;
  border: 1px solid;
  font-size: 1rem;
  display: flex;
  justify-content: center;
  background-color: var(--main-color);
}

.selected {
  background-color: var(--white);
  margin: 0.15rem;
  width: inherit;
}

.cross {
  color: var(--secondary-color);
  font-size: 4rem;
  position: relative;
  top: -17px;
}
