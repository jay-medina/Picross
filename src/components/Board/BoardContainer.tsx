import React from 'react';
import { CellState } from '../Cell';
import Board, { CellStates } from './index';

export interface BoardContainerProps {
  rows: number;
  columns: number;
}


export interface BoardContainerState {
  cellStates: CellStates;
}

class BoardContainer extends React.PureComponent<BoardContainerProps, BoardContainerState> {
  constructor() {
    super();
    this.onClick = this.onClick.bind(this);
    this.state = {
      cellStates: {},
    };
  }

  render() {
    return <Board 
      {...this.props} 
      onClick={this.onClick}
      cellStates={this.state.cellStates}
    />;
  }

  private onClick(row: number, column: number) {
    const key = `${row} - ${column}`;

    const newCellStates = {
      ...this.state.cellStates,
      [key]: this.toggleStates(
        this.state.cellStates[key],
      ),
    };

    this.setState({
      cellStates: newCellStates,
    });
  }

  private toggleStates(state?: CellState) {
    switch (state) {
      case CellState.Selected: return CellState.Crossed;
      case CellState.Crossed: return undefined;
      default: return CellState.Selected;
    }
  }

}

export default BoardContainer;
