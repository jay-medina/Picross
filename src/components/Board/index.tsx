import React from 'react';
import { Range } from 'immutable';
import Cell from '../Cell';
import BoardHints from '../BoardHints';
import { getState, createKey, CellStates } from '../Cell/cellStateTransformer';
import './index.css';

export interface BoardProps {
  rows: number;
  columns: number;
  onClick: (row: number, column: number) => void;
  cellStates: CellStates;
  rowHints: number[][];
  columnHints: number[][];
}

class Board extends React.PureComponent<BoardProps, {}> {
  render() {
    return (
      <div>
        <div className="columns">
          {this.createColumns()}
        </div>
        {this.createRows()}
      </div>
    );
  }
  private createColumns() {
    const { columnHints } = this.props;
    return Range(0, this.props.columns).map(
      (value, key) =>
        <BoardHints direction="column" key={key} hints={columnHints[value as number]} />,
    ).toArray();
  }

  private createRows() {
    const { rows } = this.props;

    return Range(0, rows).map(
      value => this.createRow(value as number),
    ).toArray();
  }

  private createRow(rowIndex: number) {
    const { columns, rowHints } = this.props;
    return (
      <div key={rowIndex} className="row">
        <BoardHints direction="row" hints={rowHints[rowIndex]} />
        {this.createCellArray(columns, rowIndex)}
      </div>
    );
  }

  private createCellArray(columns: number, row: number) {
    return Range(0, columns).map((c) => {
      const column = c as number;
      const key = createKey(row, column);
      const state = getState(this.props.cellStates, key);
      return (
        <Cell
          key={key}
          row={row}
          column={column}
          onClick={this.props.onClick}
          state={state}
        />
      );
    }).toArray();
  }
}

export default Board;
