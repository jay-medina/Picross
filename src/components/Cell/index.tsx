import React from 'react';
import './index.css';
import { CellState } from './cellStateTransformer';

export interface CellProps {
  state?: CellState;
  row: number;
  column: number;
  onClick: (row: number, column: number) => void;
}

class Cell extends React.PureComponent<CellProps, {}> {
  constructor() {
    super();
    this.onClick = this.onClick.bind(this);
  }
  render() {
    return (
      <div className="cell" onClick={this.onClick}>
        {this.getChildElement()}
      </div>
    );
  }
  private getChildElement() {
    const { state } = this.props;
    switch (state) {
      case 'crossed': return <span className="cross">&times;</span>;
      case 'selected': return <div className="selected" />;
      default: return null;
    }
  }
  private onClick() {
    this.props.onClick(this.props.row, this.props.column);
  }
}

export default Cell;
