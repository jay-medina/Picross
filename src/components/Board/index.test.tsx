import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Board from './index';
import { CellState } from '../Cell/cellStateTransformer';

describe('<Board />', () => {
  let wrapper: ShallowWrapper;

  describe('when user creates 2x2 board', () => {
    beforeEach(() => {
      wrapper = shallow(
        <Board rows={2} columns={2} onClick={jest.fn()} cellStates={{}}/>,
      );
    });

    it('display the board', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when user has a selected cell', () => {
    beforeEach(() => {
      const cellStates = {
        '1 - 1': CellState.Selected,
      };
      wrapper = shallow(
        <Board rows={2} columns={2} onClick={jest.fn()} cellStates={cellStates}/>,
      );
    });

    it('display the board with the cell selected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when a cell is clicked', () => {
    let onClick: jest.Mock<void>;

    beforeEach(() => {
      onClick = jest.fn();

      const cellStates = {
        '1 - 1': CellState.Selected,
      };
      wrapper = shallow(
        <Board rows={2} columns={2} onClick={onClick} cellStates={cellStates}/>,
      );

      wrapper.find('Cell').at(1).simulate('click', 0, 1);
    });
    it('triggers the on click', () => {
      expect(onClick).toHaveBeenCalledWith(0, 1);
    });
  });
});
