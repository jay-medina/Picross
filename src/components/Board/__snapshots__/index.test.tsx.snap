// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> when user creates 2x2 board display the board 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Board
    cellStates={Object {}}
    columns={2}
    onClick={[Function]}
    rows={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="row"
        >
          <BoardHints
            direction="row"
            hints={
              Array [
                1,
                2,
                3,
              ]
            }
          />
          <Cell
            column={0}
            onClick={[Function]}
            row={0}
            state={undefined}
          />
          <Cell
            column={1}
            onClick={[Function]}
            row={0}
            state={undefined}
          />
        </div>,
        <div
          className="row"
        >
          <BoardHints
            direction="row"
            hints={
              Array [
                1,
                2,
                3,
              ]
            }
          />
          <Cell
            column={0}
            onClick={[Function]}
            row={1}
            state={undefined}
          />
          <Cell
            column={1}
            onClick={[Function]}
            row={1}
            state={undefined}
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <BoardHints
              direction="row"
              hints={
                Array [
                  1,
                  2,
                  3,
                ]
              }
            />,
            Array [
              <Cell
                column={0}
                onClick={[Function]}
                row={0}
                state={undefined}
              />,
              <Cell
                column={1}
                onClick={[Function]}
                row={0}
                state={undefined}
              />,
            ],
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "direction": "row",
              "hints": Array [
                1,
                2,
                3,
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "0 - 0",
            "nodeType": "class",
            "props": Object {
              "column": 0,
              "onClick": [Function],
              "row": 0,
              "state": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "0 - 1",
            "nodeType": "class",
            "props": Object {
              "column": 1,
              "onClick": [Function],
              "row": 0,
              "state": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <BoardHints
              direction="row"
              hints={
                Array [
                  1,
                  2,
                  3,
                ]
              }
            />,
            Array [
              <Cell
                column={0}
                onClick={[Function]}
                row={1}
                state={undefined}
              />,
              <Cell
                column={1}
                onClick={[Function]}
                row={1}
                state={undefined}
              />,
            ],
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "direction": "row",
              "hints": Array [
                1,
                2,
                3,
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1 - 0",
            "nodeType": "class",
            "props": Object {
              "column": 0,
              "onClick": [Function],
              "row": 1,
              "state": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1 - 1",
            "nodeType": "class",
            "props": Object {
              "column": 1,
              "onClick": [Function],
              "row": 1,
              "state": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row"
          >
            <BoardHints
              direction="row"
              hints={
                Array [
                  1,
                  2,
                  3,
                ]
              }
            />
            <Cell
              column={0}
              onClick={[Function]}
              row={0}
              state={undefined}
            />
            <Cell
              column={1}
              onClick={[Function]}
              row={0}
              state={undefined}
            />
          </div>,
          <div
            className="row"
          >
            <BoardHints
              direction="row"
              hints={
                Array [
                  1,
                  2,
                  3,
                ]
              }
            />
            <Cell
              column={0}
              onClick={[Function]}
              row={1}
              state={undefined}
            />
            <Cell
              column={1}
              onClick={[Function]}
              row={1}
              state={undefined}
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <BoardHints
                direction="row"
                hints={
                  Array [
                    1,
                    2,
                    3,
                  ]
                }
              />,
              Array [
                <Cell
                  column={0}
                  onClick={[Function]}
                  row={0}
                  state={undefined}
                />,
                <Cell
                  column={1}
                  onClick={[Function]}
                  row={0}
                  state={undefined}
                />,
              ],
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "direction": "row",
                "hints": Array [
                  1,
                  2,
                  3,
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "0 - 0",
              "nodeType": "class",
              "props": Object {
                "column": 0,
                "onClick": [Function],
                "row": 0,
                "state": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "0 - 1",
              "nodeType": "class",
              "props": Object {
                "column": 1,
                "onClick": [Function],
                "row": 0,
                "state": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <BoardHints
                direction="row"
                hints={
                  Array [
                    1,
                    2,
                    3,
                  ]
                }
              />,
              Array [
                <Cell
                  column={0}
                  onClick={[Function]}
                  row={1}
                  state={undefined}
                />,
                <Cell
                  column={1}
                  onClick={[Function]}
                  row={1}
                  state={undefined}
                />,
              ],
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "direction": "row",
                "hints": Array [
                  1,
                  2,
                  3,
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1 - 0",
              "nodeType": "class",
              "props": Object {
                "column": 0,
                "onClick": [Function],
                "row": 1,
                "state": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1 - 1",
              "nodeType": "class",
              "props": Object {
                "column": 1,
                "onClick": [Function],
                "row": 1,
                "state": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Board /> when user has a selected cell display the board with the cell selected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Board
    cellStates={
      Object {
        "1 - 1": "selected",
      }
    }
    columns={2}
    onClick={[Function]}
    rows={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="row"
        >
          <BoardHints
            direction="row"
            hints={
              Array [
                1,
                2,
                3,
              ]
            }
          />
          <Cell
            column={0}
            onClick={[Function]}
            row={0}
            state={undefined}
          />
          <Cell
            column={1}
            onClick={[Function]}
            row={0}
            state={undefined}
          />
        </div>,
        <div
          className="row"
        >
          <BoardHints
            direction="row"
            hints={
              Array [
                1,
                2,
                3,
              ]
            }
          />
          <Cell
            column={0}
            onClick={[Function]}
            row={1}
            state={undefined}
          />
          <Cell
            column={1}
            onClick={[Function]}
            row={1}
            state="selected"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <BoardHints
              direction="row"
              hints={
                Array [
                  1,
                  2,
                  3,
                ]
              }
            />,
            Array [
              <Cell
                column={0}
                onClick={[Function]}
                row={0}
                state={undefined}
              />,
              <Cell
                column={1}
                onClick={[Function]}
                row={0}
                state={undefined}
              />,
            ],
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "direction": "row",
              "hints": Array [
                1,
                2,
                3,
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "0 - 0",
            "nodeType": "class",
            "props": Object {
              "column": 0,
              "onClick": [Function],
              "row": 0,
              "state": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "0 - 1",
            "nodeType": "class",
            "props": Object {
              "column": 1,
              "onClick": [Function],
              "row": 0,
              "state": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <BoardHints
              direction="row"
              hints={
                Array [
                  1,
                  2,
                  3,
                ]
              }
            />,
            Array [
              <Cell
                column={0}
                onClick={[Function]}
                row={1}
                state={undefined}
              />,
              <Cell
                column={1}
                onClick={[Function]}
                row={1}
                state="selected"
              />,
            ],
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "direction": "row",
              "hints": Array [
                1,
                2,
                3,
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1 - 0",
            "nodeType": "class",
            "props": Object {
              "column": 0,
              "onClick": [Function],
              "row": 1,
              "state": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1 - 1",
            "nodeType": "class",
            "props": Object {
              "column": 1,
              "onClick": [Function],
              "row": 1,
              "state": "selected",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row"
          >
            <BoardHints
              direction="row"
              hints={
                Array [
                  1,
                  2,
                  3,
                ]
              }
            />
            <Cell
              column={0}
              onClick={[Function]}
              row={0}
              state={undefined}
            />
            <Cell
              column={1}
              onClick={[Function]}
              row={0}
              state={undefined}
            />
          </div>,
          <div
            className="row"
          >
            <BoardHints
              direction="row"
              hints={
                Array [
                  1,
                  2,
                  3,
                ]
              }
            />
            <Cell
              column={0}
              onClick={[Function]}
              row={1}
              state={undefined}
            />
            <Cell
              column={1}
              onClick={[Function]}
              row={1}
              state="selected"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <BoardHints
                direction="row"
                hints={
                  Array [
                    1,
                    2,
                    3,
                  ]
                }
              />,
              Array [
                <Cell
                  column={0}
                  onClick={[Function]}
                  row={0}
                  state={undefined}
                />,
                <Cell
                  column={1}
                  onClick={[Function]}
                  row={0}
                  state={undefined}
                />,
              ],
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "direction": "row",
                "hints": Array [
                  1,
                  2,
                  3,
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "0 - 0",
              "nodeType": "class",
              "props": Object {
                "column": 0,
                "onClick": [Function],
                "row": 0,
                "state": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "0 - 1",
              "nodeType": "class",
              "props": Object {
                "column": 1,
                "onClick": [Function],
                "row": 0,
                "state": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <BoardHints
                direction="row"
                hints={
                  Array [
                    1,
                    2,
                    3,
                  ]
                }
              />,
              Array [
                <Cell
                  column={0}
                  onClick={[Function]}
                  row={1}
                  state={undefined}
                />,
                <Cell
                  column={1}
                  onClick={[Function]}
                  row={1}
                  state="selected"
                />,
              ],
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "direction": "row",
                "hints": Array [
                  1,
                  2,
                  3,
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1 - 0",
              "nodeType": "class",
              "props": Object {
                "column": 0,
                "onClick": [Function],
                "row": 1,
                "state": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1 - 1",
              "nodeType": "class",
              "props": Object {
                "column": 1,
                "onClick": [Function],
                "row": 1,
                "state": "selected",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
